# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1312411139 25200
#      Wed Aug 03 15:38:59 2011 -0700
# Node ID 9e1b9ca119eb67518ccb0eb1fc26175085f0a0bb
# Parent  ac00ae83ea2eba7d0a1ee65670334ed6e157233f
strsplit.m: Allow any scalar input for strip_empty option,
not just logical value.

* strsplit.m: Allow any scalar input for strip_empty option,
not just logical value.

diff --git a/scripts/strings/strsplit.m b/scripts/strings/strsplit.m
--- a/scripts/strings/strsplit.m
+++ b/scripts/strings/strsplit.m
@@ -23,17 +23,17 @@
 ## boundaries result in empty strings, unless @var{strip_empty} is true.
 ## The default value of @var{strip_empty} is false.
 ## @seealso{strtok}
 ## @end deftypefn
 
 function s = strsplit (p, sep, strip_empty = false)
 
   if (nargin < 2 || nargin > 3 || ! ischar (p) || rows (p) > 1
-      || ! ischar (sep) || ! islogical (strip_empty))
+      || ! ischar (sep) || ! isscalar (strip_empty))
     print_usage ();
   endif
 
   if (isempty (p))
     s = cell (size (p));
   else
     ## Split p according to delimiter.
     if (isscalar (sep))
